src/etl/permits_extractor.py

```
@@ -0,0 +1,953 @@
+            processed['ingest_timestamp'] = datetime.now().isoformat()
+            
+            # Generate natural key hash
+            key_fields = ['permit_number', 'issue_date']
+            key_values = [str(processed.get(f, '')) for f in key_fields]
+            processed['natural_key_hash'] = hashlib.md5(
+                '|'.join(key_values).encode()
+            ).hexdigest()
+            
+            return processed
+            
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)